packages/kit/src/runtime/server/endpoint.js

```diff
@@ -40,28 +40,36 @@ export async function render_endpoint(event, mod, state) {
 	}
 
 	try {
-		let response = await handler(
-			/** @type {import('@sveltejs/kit').RequestEvent<Record<string, any>>} */ (event)
-		);
-
-		if (!(response instanceof Response)) {
-			throw new Error(
-				`Invalid response from route ${event.url.pathname}: handler should return a Response object`
+		if (method === 'GET' && event.request.headers.has('upgrade') && event.upgrade && mod.UPGRADE) {
+			console.log('upgrade');
+			await mod.UPGRADE(/** @type {import('@sveltejs/kit').RequestEvent<Record<string, any>>} */ (event));
```

Comment:
if you put a `return` after this line you could avoid wrapping everything that follows in an `else`

Reply:
Should I just simplify it to a conditional return?